bi_tools.py

```diff
@@ -1,58 +1,174 @@
-from Functions import run_dna_rna_tools_funcs as seqf
-from Functions import fastq_filter_funcs as fastqf
+from abc import ABC, abstractmethod
+from Bio import SeqIO
+from Bio.SeqUtils import GC
 
 
-def run_dna_rna_tools(*args: tuple):
-    '''
-    Function run_dna_rna_tools, makes transformation
-    specified in the last argument
+class BiologicalSequence(ABC):
+    def __init__(self, sequence):
+        self.sequence = sequence
 
-    Args: tuple
+    @abstractmethod
+    def get_length(self) -> int:
+        pass
 
-    Returns: str or list or float
-    '''
-    result = list()
-    seqs = list(args[:-1])
-    for seq in seqs:
-        if args[-1] == "reverse":
-            result.append(seqf.reverse(seq))
-        if args[-1] == "complement":
-            result.append(seqf.complement(seq))
-        if args[-1] == "reverse_complement":
-            result.append(seqf.reverse_complement(seq))
-        if args[-1] == "transcribe":
-            result.append(seqf.transcribe(seq))
-        if args[-1] == "g_c_bound":
-            result.append(seqf.get_g_c_bound(seq))
-    if len(result) == 1:
-        return result[0]
-    else:
-        return result
+    @abstractmethod
+    def get_subsequence(self, start: int, end: int) -> str:
+        pass
+
+    @abstractmethod
+    def print_sequence(self) -> None:
+        pass
+
+    @abstractmethod
+    def check_alphabet(self, sequence: str) -> bool:
+        pass
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    def __init__(self, sequence: str):
+        super().__init__(sequence)
+
+    def get_length(self) -> int:
+        return len(self.sequence)
+
+    def get_subsequence(self, start, end) -> str:
+        return self.sequence[start:end+1]
+
+    def print_sequence(self) -> None:
+        print(self.sequence)
+
+    def complement(self) -> str:
+        '''
+        Function complement, transforms initial sequence to complemented
+
+        Args: str
+
+        Returns: str
+        '''
+        init_line = "ACGTUacgtu"
+        exit_line = "TGCAAtgcaa"
+        tab = str(self.sequence).maketrans(init_line, exit_line)
+        return str(self.sequence).translate(tab)
+
+    def reverse(self) -> str:
+        '''
+        Function reverse, transforms initial sequence to reversed
+
+        Args: str
+
+        Returns: str
+        '''
+        return str(self.sequence[::-1])
+
+    def reverse_complement(self) -> str:
+        '''
+        Function reverse_complement, transforms initial sequence to reversed and complemented
+
+        Args: str
+
+        Returns: str
+        '''
+        self.sequence = self.reverse()
+        rev_compl_seq = self.complement()
+        return ''.join(rev_compl_seq)
+
+    def get_g_c_score(self) -> float:
```

Comment:
```suggestion
    def get_gc_score(self) -> float:
```
Это уже общепринятая биологическая штука, не стоит gc разделять.